ui/selectors/transactions.js

```diff
@@ -21,6 +22,20 @@ const INVALID_INITIAL_TRANSACTION_TYPES = [
   TransactionType.retry,
 ];
 
+// The statuses listed below are permitted in the Activity list for Smart Swaps because
+// a Smart Swap with these statuses hasn't been added to the TransactionController's regular transaction list yet.
+// The SUCCESS status is not allowed because once a Smart Swap is successful,
+// the "this.confirmExternalTransaction" function is called, which moves the transaction
+// to the regular transaction list in the TransactionController.
+// TODO: When Swaps and non-Swaps transactions are treated the same,
+// we will only allow the PENDING smart transaction status in the Activity list.
+const allowedSwapsSmartTransactionStatusesForActivityList = [
+  SmartTransactionStatuses.PENDING,
+  SmartTransactionStatuses.UNKNOWN,
+  SmartTransactionStatuses.RESOLVED,
+  SmartTransactionStatuses.CANCELLED, // Other CANCELLED_ statuses are deprecated.
```

Comment:
Instead of putting it here, let's add the comment about deprecated statuses to the original enum in the smart transaction controller repo. We can do this in a separate PR.